.github/workflows/legacy-npm-workflow.yml

```diff
@@ -0,0 +1,191 @@
+name: Build and pack Npm components
+on:
+  workflow_call:
+    inputs:
+      deploy:
+        required: false
+        type: string
+        default: 'false'
+      run_tests:
+        required: false
+        type: string
+        default: 'true'
+      npm_version:
+        required: false
+        type: string
+        default: '16.x'
+
+    secrets:
+      auto_commit_user:
+        required: true
+      auto_commit_mail:
+        required: true
+      auto_commit_pwd:
+        required: true
+      registry_user:
+        required: true
+      registry_password:
+        required: true
+      gh_token:
+        required: true
+      sonar_token:
+        required: true
+      npm_token:
+        required: true
+
+jobs:
+  Build:
+
+    runs-on: ubuntu-20.04
+    container:
+      image: registry.aws.cloud.mov.ai/devops/npm-buildserver:latest
+      options: --user root
+      credentials:
+        username: ${{secrets.registry_user}}
+        password: ${{secrets.registry_password}}
+    env:
+      NODE_AUTH_TOKEN: ${{ secrets.npm_token }}
+
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        # Disabling shallow clone is recommended for improving relevancy of reporting
+        fetch-depth: 0
+        submodules: recursive
+        token: ${{ secrets.npm_token }}
+
+    - name: Raise App version
+      run: |
+        git config --global --add safe.directory $(pwd)
+        git config --global user.name '${{ secrets.auto_commit_user }}'
+        git config --global user.email '${{ secrets.auto_commit_mail }}'
+        git config --global user.password ${{ secrets.auto_commit_pwd }}
+        npm version prerelease --no-git-tag-version
+
+    - name: Setup Github as proxy Registry
+      uses: actions/setup-node@v2
+      with:
+        node-version: '${{ inputs.npm_version }}'
+        registry-url: 'https://npm.pkg.github.com/mov-ai'
+
+    - name: Install dependencies
+      run: npm ci --loglevel verbose --legacy-peer-deps
```

Comment:
@duartecoelhomovai shouldn't we add a parameter on the workflow : `legacy: true/false` for better maintainability

Reply:
That sounds awesome

Reply:
sure. will do that before the end of the day :) will let you know once i have a minute to add the if and input argument.

Reply:
I tried saving you some work. Please take a look

Reply:
looks good to me

Reply:
@quirinpa  please confirm it runs OK with maybe a run on https://github.com/MOV-AI/frontend-npm-lib-react/pull/267 where you specify `@legacy-deps` instead of `@v2`

Reply:
Confirmed